src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -336,11 +336,22 @@ public function reset()
      *
      * @return array An array of all defined service ids
      */
-    public function getServiceIds()
+    public function getServiceIds(/*$includePrivates = true*/)
     {
+        $includePrivates = true;
+        if (func_num_args() === 1) {
```

Comment:
Nobody calls this with an argument, which means nobody will see the deprecation notice.
Yet, I think we should only add a note to remind about not returning private services in 4.0. No need for any deprecations: if people use the private services returned today from this method, they will get a deprecation because fetching them is deprecated.